### PR TITLE
Tslint to eslint

### PR DESCRIPTION
**Pull request by kian2attari**
###  Summary

This PR aims to migrate Bolt.js from TSLint to ESLint. The change was done individually for each package in the monorepo in order to maintain the structure. Currently, this PR includes a complete a migration to ESLint along with the `typescript-eslint` plugin, the `airbnb-typescript` config, and `prettier` integration. Similar to the TSLint->ESLint migration for Bolt-js, `prettier` in run in place of ESLint's formatting rules (ex. indent rules) as is recommended by the `typescript-eslint` team.  The `lint` command now first runs prettier to make sure all formatting is correct before eslint is run. 

Since the AirBnB standard has changed a bit since it was used in this project's original tsconfig, there are a few eslint rules that currently fail (similar to the situation with the Bolt-js PR). The two options now are to either disable these new rules, or make the fixes necessary to conform to them. As the rules are fairly easy to conform to (many of them even auto fixes), I'm going to go the path of applying the necessary fixes.

**Note: Any changes in project files resulted from me running the `prettier-fix` script  (in `package.json`) on all files just to get consistent formatting. They are purely formatting changes rather than syntax changes.**

### Requirements (place an `x` in each `[ ]`)

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

File: packages/client/.vscode/settings.json [hunk 1]
```diff
@@ -1,3 +1,16 @@
...
+project's original TSLint configuration.
+
+We recommend eventually switching this configuration to extend from
+the recommended rulesets in typescript-eslint. 
+https://github.com/typescript-eslint/tslint-to-eslint-config/blob/master/docs/FAQs.md
+
+Happy linting! 💖
+*/
 {
     "typescript.tsdk": "./node_modules/typescript/lib"
```
Comment (stevengill) [inline packages/client/.vscode/settings.json:15]: In all of the settings.json, can we follow the pattern we did with bolt? https://github.com/slackapi/bolt-js/blob/main/.vscode/settings.json

Essentially turning prettier to run on save.

File: packages/client/package.json [hunk 2]
```diff
@@ -41,7 +41,8 @@
     "prepare": "npm run build",
     "build": "npm run build:clean && tsc",
     "build:clean": "shx rm -rf ./dist",
-    "lint": "tslint --project .",
+    "lint": "prettier 'src/**/*.ts' --check && eslint --ext .ts src",
+    "prettier-fix": "prettier --write 'src/**/*.ts'",
```
Comment (stevengill) [inline packages/client/package.json:45]: nice! maybe we rename this to `lint-fix` and make it run eslint with the fix flag too.

File: packages/client/package.json [hunk 3]
```diff
@@ -52,6 +53,14 @@
...
+    "@typescript-eslint/eslint-plugin": "^3.9.0",
+    "@typescript-eslint/parser": "^3.9.0",
+    "eslint": "^7.6.0",
+    "eslint-config-airbnb-typescript": "^9.0.0",
+    "eslint-config-prettier": "^6.11.0",
+    "eslint-plugin-import": "^2.22.0",
+    "eslint-plugin-jsdoc": "^30.2.1",
+    "prettier": "^2.0.5",
     "shx": "^0.3.2",
     "tslint": "^5.13.1",
```
Comment (stevengill) [inline packages/client/package.json:65]: can we remove tslint as a dep?

**Comment by stevengill** [comment]
@kian2attari can you rebase from main? Maybe the CI tests will pass now.

**Comment by seratch** [comment]
@stevengill TSLint is already deprecated two years ago. Let's discuss the quick way to move the migration forward 🚀

**Comment by filmaj** [comment]
Closing in favour of #1325.

**Pull request closed without merging**